test/commands/github/repository/update_branch_protection_test.rb

```diff
@@ -0,0 +1,406 @@
+require "test_helper"
+
+class Github::Repository::UpdateBranchProtectionTest < ActiveSupport::TestCase
+  test "sets correct branch protection for active track's repo" do
+    create :track, slug: 'elixir', active: true
+
+    stub_request(:get, "https://api.github.com/repos/exercism/elixir/branches/main/protection").
+      to_return(status: 200, body: current_branch_protection_settings.to_json, headers: { 'Content-Type': 'application/json' })
+
+    stub_request(:put, "https://api.github.com/repos/exercism/elixir/branches/main/protection").
+      with(body: hash_including({
```

Comment:
Just found out this `hash_including` method to partially match the request body. It is great!